### PR TITLE
update mongoose to v6 and use mongoose-v6 QueryWithHelpers interface

### PR DESCRIPTION
**Pull request by eSRoye**
Update dependency mongoose to v6 and use mongoose-v6 QueryWithHelpers interface instead of DocumentQuery interface which is not present anymore.

### CODE REVIEW

File: packages/casl-mongoose/package.json [hunk 1]
```diff
@@ -39,14 +39,14 @@
   "license": "MIT",
   "peerDependencies": {
     "@casl/ability": "^3.0.0 || ^4.0.0 || ^5.1.0",
-    "mongoose": "^5.12.0"
+    "mongoose": "^6.0.0"
```
[the remark this answers is not in the document]
Comment (brano93) [reply packages/casl-mongoose/package.json:42]: @stalniy Is this going to be merged any time soon?

**Comment by stalniy** [comment]
looks fine. the only thing which is left is to update documentation to reflect this latest change

**Comment by stalniy** [comment]
OK, checked the docs it looks like everything should be fine without changes. I will double-check the example with the latest changes and will release it today.

PS: sorry for such a long delay guys

**Comment by stalniy** [comment]
released in @casl/mongoose@6.0.0

**Pull request merged**